### DISCUSSION TITLE
Consider alternatives to overlayroot

### DISCUSSION THREAD

**Discussion by UriHerrera** [Nitrux/nitrux-bug-tracker#135] [Ideas]
With Linux 6.5, overlayroot does not work anymore; see https://github.com/Nitrux/nitrux-bug-tracker/issues/133.

I've no idea why or how. All I know is that it only works with kernels up to 6.4.15. Unfortunately, this isn't a trivial matter since the [Nitrux Update Tool System](https://github.com/Nitrux/nuts/) is affected, too.

I've searched for alternatives. However, there's no natural alternative on the horizon.

1. OSTree does not support initramfs-tools, only dracut, and it's highly dependent on Systemd and, as far as I can tell, BTRFS; see rpm-ostree for example.
2. Debian "supports" dracut as much as it "supports" OpenRC, so there's that. In other words, there is a package, but anything that uses initramfs hooks targets initramfs-tools (so, literally everything in Debian).
3. I found a project called [fsprotect](https://github.com/sharhalakis/fsprotect); however, it depends on AUFS.
4. I also found another project called [Bilibop](https://un.poivron.org/~quidame/wiki/bilibop/), which depends on AUFS too.
5. Debian **does not support AUFS**, so there's that. Specifically, Debian does have packages for AUFS version 5.2, which is no longer supported by its developer, and it does not have support for newer kernels because [AUFS](https://aufs.sourceforge.net/) version 6 is the latest.

**Comment by UriHerrera** [comment]
I've updated the original issue: https://github.com/Nitrux/nitrux-bug-tracker/issues/133.

TL;DR 

_Seemingly_ a change in how the kernel parameter `ro` works causes the root to be read-only. It is the purpose of that parameter to do that precisely per its documentation; however, using the same parameter in previous kernels (pre-6.5) didn't cause an adverse effect.

When I remove the parameter and boot, there are no problems, and the overlay works as expected. So, the root remains immutable even if I use a newer kernel, in this case, 6.5.11-1.

Therefore, I'll change the title of this discussion as overlayroot *does* work, but that parameter is causing the issue. Nonetheless, I don't know _what_ adds that parameter.

**Reply by UriHerrera** [reply]
I've made some changes that will take care of removing the parameter. However, for future-proofing, we should look for an alternative to overlayroot.

Maybe implement some of the things that znx did well.

**Comment by UriHerrera** [comment]
I've forked and rewritten overlayroot. 

https://github.com/Nitrux/overlayroot
